user_events.py

```diff
@@ -205,9 +205,17 @@ def user_an_event(id):
                     subEvent['isPublished'] = True
                 else:
                     subEvent['isPublished'] = False
-            else:
+            elif 'eventid' in subEvent:
                 event = find_user_event(subEvent['eventid'])
                 subEvent['isPublished'] = False
+            else:
+                post['subEvents'].remove(subEvent)
+                print("remove illegal subevent")
```

Comment:
```suggestion
                _logger.debug("remove illegal subevent")
```
Same comment as previous: 
Could we use `_logger.debug` statement here? Also, illegal might not be the appropriate word. Are we trying to say deleted subevent?